### PR TITLE
Add Domain Layer

### PR DESCRIPTION
**Pull request by hassanSayed0**
- create package Domain
 - create file FetchUserUseCase to control core layer and convert userCodable to User
 - create file user and extend userCodable to add function toUser

### CODE REVIEW

File: Domain/.gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.DS_Store
+/.build
```
Comment (Mahmoud-Bassuni) [inline Domain/.gitignore:2]: not needed this file as we have one on e-store

File: Domain/Sources/Domain/UseCase/FetchUserUseCase.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
+//
+//  File.swift
+//  
+//
+//  Created by Hassan on 23/03/2023.
+//
+
+import Foundation
+import Core
+public struct FetchUserUseCase {
```
Comment (Mahmoud-Bassuni) [inline Domain/Sources/Domain/UseCase/FetchUserUseCase.swift:10]: rename it to UserUseCase

File: Domain/Sources/Domain/UseCase/FetchUserUseCase.swift [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        userRepo.fetchUsers { result in
+            switch result {
+            case .success(let usersCodable):
+                completion(usersCodable.toUser())
+            case .failure(let error):
+                print("failure \(error)")
+            }
+        }
+    }
+    public func fetchTokenLoginUser(loginInfo localLoginInfo: (userName:String,passWord:String), completion: @escaping (Token) -> Void) {
```
Comment (Mahmoud-Bassuni) [inline Domain/Sources/Domain/UseCase/FetchUserUseCase.swift:34]: rename it to Login

File: Domain/Sources/Domain/extension/ArrayExtension.swift [hunk 4]
```diff
@@ -0,0 +1,16 @@
+//
+//  File.swift
+//  
+//
+//  Created by Hassan on 23/03/2023.
+//
+
+import Foundation
+import Core
```
Comment (Mahmoud-Bassuni) [inline Domain/Sources/Domain/extension/ArrayExtension.swift:9]: plz rename the file to UserListMapper

File: Domain/Sources/Domain/extension/UserCodableExtension.swift [hunk 5]
```diff
@@ -0,0 +1,34 @@
+//
+//  File.swift
+//  
+//
+//  Created by Hassan on 25/03/2023.
+//
+
+import Foundation
+import Core
+extension UserCodable {
```
Comment (Mahmoud-Bassuni) [inline Domain/Sources/Domain/extension/UserCodableExtension.swift:10]: plz don't use extension UserCodable you can create new struct UserMapper and create func to map user from codable to User

File: Domain/Sources/Domain/extension/UserMapper.swift [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+//  File.swift
+//  
+//
+//  Created by Hassan on 25/03/2023.
+//
+
+import Foundation
+import Core
+public struct UserMapper {
+    public func userCodableToUser(_ inuser: UserCodable) -> User {
```
Comment (Mahmoud-Bassuni) [inline Domain/Sources/Domain/extension/UserMapper.swift:11]: rename to transformCodableToUser ,
 rename inuser to model 
and rename folder extension to Mapper

Comment (hassanSayed0) [reply Domain/Sources/Domain/extension/UserMapper.swift]: done

**Review by Mahmoud-Bassuni — APPROVED** [review]

**Review by AhmedAtef277 — APPROVED** [review]

**Pull request merged**